packages/x-date-pickers/src/MonthCalendar/MonthCalendar.types.ts

```diff
@@ -1,8 +1,23 @@
+import * as React from 'react';
 import { SxProps } from '@mui/system';
 import { Theme } from '@mui/material/styles';
+import { SlotComponentProps } from '@mui/base/utils';
 import { MonthCalendarClasses } from './monthCalendarClasses';
 import { BaseDateValidationProps, MonthValidationProps } from '../internals/models/validation';
 import { PickerValidDate, TimezoneProps } from '../models';
+import type { PickersMonthProps } from './PickersMonth';
+
+export interface MonthCalendarSlots {
```

Comment:
I'm creating the slots here and not in `PickersMonth` since `PickersMonth` will disappear in v8